sklearn/tree/_tree.pyx

```diff
@@ -203,14 +203,18 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
         cdef intp_t parent
         cdef bint is_left
         cdef intp_t n_node_samples = splitter.n_samples
-        cdef double weighted_n_node_samples
+        cdef float64_t weighted_n_node_samples
         cdef SplitRecord split
         cdef intp_t node_id
 
-        cdef double impurity = INFINITY
-        cdef double lower_bound
-        cdef double upper_bound
-        cdef double middle_value
+        cdef float64_t impurity = INFINITY
+        cdef float64_t lower_bound
+        cdef float64_t upper_bound
+        cdef float64_t middle_value
+        cdef float64_t left_child_min
+        cdef float64_t left_child_max
+        cdef float64_t right_child_min
+        cdef float64_t right_child_max
```

Comment:
Good catch! Those were not defined before. I looked into _tree.cpp and obviously Cython figures out to define them as double variables.